eth2/operation_pool/src/lib.rs

```diff
@@ -118,6 +119,9 @@ impl<T: EthSpec> OperationPool<T> {
         let prev_domain_bytes = AttestationId::compute_domain_bytes(prev_epoch, state, spec);
         let curr_domain_bytes = AttestationId::compute_domain_bytes(current_epoch, state, spec);
         let reader = self.attestations.read();
+        let validator_statuses = ValidatorStatuses::new(state, spec)
+            .expect("should have returned valid validator statuses");
+        let total_active_balance = validator_statuses.total_balances.current_epoch;
```

Comment:
Rather than computing the whole validator statuses object here, it would be quicker to compute the total active balance using `BeaconState::get_total_balance` and `BeaconState::get_cached_active_validator_indices`.

Reply:
We should also avoid calling `expect` here. It's strange that there's no error type for the op pool, we should probably make one, something like:

```rust
enum OpPoolError {
    GetAttestationsTotalBalanceError(BeaconStateError)
    // .. other variants ..
}
```